internal/node/preflight/init.go

```diff
@@ -16,39 +17,51 @@ var (
 	magenta = color.New(color.FgMagenta).SprintFunc()
 	green   = color.New(color.FgHiGreen).SprintFunc()
 
-	nexLatestVersion = func(ctx context.Context) string {
+	nexLatestVersion = func(ctx context.Context, logger *slog.Logger) string {
 		version := "development"
 		build_data, ok := ctx.Value("build_data").(map[string]interface{})
 		if ok {
 			version, ok = build_data["version"].(string)
 			if !ok {
-				fmt.Println("error parsing version from build data")
+				logger.Warn("error parsing version from build data")
 			}
 		}
 		if version == "development" {
+			backoff := 0
+
+		RETRY:
 			res, err := http.Get("https://api.github.com/repos/synadia-io/nex/releases/latest")
 			if err != nil {
-				fmt.Printf("error making http request: %s\n", err)
+				logger.Error("error making http request", slog.Any("err", err))
 				return ""
 			}
 			defer res.Body.Close()
+			if res.StatusCode != 200 && backoff < 5 {
+				slog.Warn("failed to get latest version, incrementing backoff", slog.String("status_code", res.Status), slog.Int("backoff", backoff))
+				backoff++
+				time.Sleep(time.Duration(backoff) * time.Second)
```

Comment:
is the maximum backoff failure time here (15 secs) going to potentially mess up CI testing?

Reply:
we are hitting a timeout....so its necessary

Reply:
cool. so the backoff is to appease the CI deities rather than anger them. :+1:

Reply:
Or anyone using GitHub as an unauthenticated user.  From what I saw in the docs, it's rate limited on IP at 60 requests and hour....if you run our test suite, you use about 25 of those requests every time 🙃